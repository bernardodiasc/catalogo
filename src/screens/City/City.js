import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import config from 'config'

import Layout from 'displays/Layout'

class City extends Component {
  static defaultProps = {
    city: null
  }

  render() {
    return (
      <Layout sidebar={false}>
        {this.props.city}
        <hr />
        <Link to={`${config.PUBLIC_URL}/ubatuba/escola-exemplo`}>Escola Exemplo</Link><br/>
        <Link to={`${config.PUBLIC_URL}/ubatuba/padaria-exemplo`}>Padaria Exemplo</Link><br/>
      </Layout>
    )
  }
}

export default City
