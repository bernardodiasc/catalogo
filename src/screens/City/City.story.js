import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Docs from 'stories/containers/Docs'

import City from './City'
import CityReadme from './README.md'
storiesOf('Screens/City', module)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{CityReadme}</Docs>)
  .add('Example', () => <City />)
  .add('Minimal render', () => <City />)
