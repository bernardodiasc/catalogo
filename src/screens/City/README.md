# City

## How to use

```
import City from 'screens/City'
```

```
<City />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
