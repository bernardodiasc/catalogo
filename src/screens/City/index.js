import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { notFound } from 'actions'
import City from './City'

const mapStateToProps = store => ({
  store: store,
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    notFound,
  }, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(City))
