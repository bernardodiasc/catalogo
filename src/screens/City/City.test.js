import React from 'react'
import { shallow } from 'enzyme'

import City from './City.js'

it('renders without crashing', () => {
  const component = shallow(<City />)
  expect(component).toMatchSnapshot()
})
