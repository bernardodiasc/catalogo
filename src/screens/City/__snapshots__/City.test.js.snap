// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Layout
  headbar={true}
  notification={false}
  sidebar={false}
  toggleHeadbar={[Function]}
  toggleSidebar={[Function]}
>
  <hr />
  <Link
    replace={false}
    to="/ubatuba/escola-exemplo"
  >
    Escola Exemplo
  </Link>
  <br />
  <Link
    replace={false}
    to="/ubatuba/padaria-exemplo"
  >
    Padaria Exemplo
  </Link>
  <br />
</Layout>
`;
